### ISSUE TITLE
Would this library support secured Solr with password?

### ISSUE THREAD

**Issue by akolacca** [inoio/solrs#123]
I am using a secured solr, and i couldn't find any reference for how to connect to a secured solr. I wonder how would this be possible

**Comment by magro** [comment]
How would you do that with SolrJ?
Can you also provide more details how exactly your Solr is secured, maybe by referencing the related docs?

**Comment by maleficuss** [comment]
with solrj its something like this
`@Configuration
public class SolrConfiguration {

    @Value("${solr.host}")
    private String host;

    @Value("${solr.username}")
    private String solrUsername;

    @Value("${solr.password}")
    private String solrPassword;

    @Bean
    public SolrClient solrClient() {
        return new Http2SolrClient.Builder(host)
                .withBasicAuthCredentials(solrUsername, solrPassword)
                .build();
    }
}`

Well you can do similar with RequestInterceptor, but somehow it ignore is it, here is my example and it still says 401 from solr server

`@Component
public class SolrAuthInterceptor implements RequestInterceptor {

    @Value("${solr.username}")
    private String solrUsername;

    @Value("${solr.password}")
    private String solrPassword;
    @Override
    public <T extends SolrResponse> Future<T> interceptRequest(
            Function2<SolrServer, SolrRequest<? extends T>, Future<T>> serverRequestLambda,
            SolrServer solrServer,
            SolrRequest<? extends T> request
    ) {
        return serverRequestLambda.apply(solrServer, request.setBasicAuthCredentials(solrUsername, solrPassword));
    }`

`@Configuration
@RequiredArgsConstructor
public class SolrClientConfiguration {

    private final SolrAuthInterceptor solrAuthInterceptor;

    @Value("${solr.host}")
    private String host;

    @Bean
    public JavaAsyncSolrClient solrClient() {
        return JavaAsyncSolrClient.builder(host)
                .withRequestInterceptor(solrAuthInterceptor)
                .build();
    }
}`

**Comment by magro** [comment]
Thanks for your input, @maleficuss!

The solution should be to use a custom http client, like this:

```scala
import org.asynchttpclient.DefaultAsyncHttpClient
import org.asynchttpclient.DefaultAsyncHttpClientConfig
import org.asynchttpclient.Realm
import org.asynchttpclient.Realm.AuthScheme

val httpClient = {
  val config = new DefaultAsyncHttpClientConfig.Builder()
  val realm = new Realm.Builder(username, password)
    .setUsePreemptiveAuth(true)
    .setScheme(AuthScheme.BASIC)
    .build()
  new DefaultAsyncHttpClient(config.setRealm(realm).build)
}

val solr: AsyncSolrClient[Future] = AsyncSolrClient.Builder(s"http://localhost:${solrRunner.port}/solr/collection1")
  .withHttpClient(httpClient)
  .build
```

I've verified this in #162 in case you'd like to see more.

**Comment by magro** [comment]
Closed with #162
